components/service-operator/controllers/serviceaccount.go

```diff
@@ -170,9 +170,7 @@ func (r *ServiceAccountController) reconcileServiceAccountWithContext(ctx contex
 		return fmt.Errorf("principal not ready")
 	}
 
-	sa.Annotations = map[string]string{
-		"eks.amazonaws.com/role-arn": principal.Status.AWS.Info[access.IAMRoleArnOutputName],
-	}
+	sa.Annotations["eks.amazonaws.com/role-arn"] = principal.Status.AWS.Info[access.IAMRoleArnOutputName]
```

Comment:
writing to this map without first checking if it is `nil` looks dangerous. It might be fine, but I don't think anything is documented guarenteeing that the map is initialized after a Get from the kubernetes api, so probably better to play it safe to avoid panic:

```
if sa.Anonotations == nil {
  sa.Annotations = map[string]string{}
}
sa.Annotations["blah"] = "blah"
```